### PR TITLE
Script security - password protection

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Feature to protect keys generated by the script.

IMPORTANT: In order to be merged, this PR requires a version of the trader that supports a version of Open Autonomy including this fix: https://github.com/valory-xyz/open-autonomy/pull/2108

### CODE REVIEW

File: run_service.sh [hunk 1]
```diff
@@ -206,6 +270,10 @@ create_storage() {
     echo "This is the first run of the script. The script will generate new operator and agent instance addresses."
     echo ""
 
+    echo "You can use a password to encrypt the generated key files. You will be asked for the password each time the script is run."
+    ask_confirm_password
+    echo ""
```
Comment (Adamantios) [inline run_service.sh:275]: These echo should be in the method no?

Comment (jmoreira-valory) [reply run_service.sh:275]: fixed

**Review by Adamantios — APPROVED** [review]

File: run_service.sh [hunk 2]
```diff
@@ -680,15 +752,21 @@ else
...
-poetry run autonomy deploy build --n $n_agents -ltm
+poetry run autonomy deploy build "../../$keys_json_path" --n $n_agents -ltm $password_argument
 
 cd ..
 
 add_volume_to_service "$PWD/trader_service/abci_build/docker-compose.yaml" "trader_abci_0" "/data" "$PWD/../.trader_runner/"
 
 # Run the deployment
-poetry run autonomy deploy run --build-dir $directory --detach
+poetry run autonomy deploy run --build-dir $directory --detach || true
```
Comment (jmoreira-valory) [inline run_service.sh:765]: true is needed to proceed to clear the password stored by the framework, even if this command fails.

**Comment by jmoreira-valory** [comment]
Bug https://github.com/valory-xyz/open-autonomy/issues/2088 has been fixed on https://github.com/valory-xyz/open-autonomy/pull/2108
Bug https://github.com/valory-xyz/open-autonomy/issues/2089 has been fixed on https://github.com/valory-xyz/open-autonomy/pull/2108

File: run_service.sh (hunk 2 above)
Comment (jmoreira-valory) [reply run_service.sh]: Not needed anymore.

**Review by Adamantios — APPROVED** [review]

**Review by Adamantios — CHANGES_REQUESTED** [review]
Tested this and the password is not enforced. I do not agree with releasing this as is.

**Comment by jmoreira-valory** [comment]
> Tested this and the password is not enforced. I do not agree with releasing this as is.

Please clarify what enforcements and where.

**Review by Adamantios — APPROVED** [review]
Tested this again. The script just fails later and not immediately.

**Comment by jmoreira-valory** [comment]
> Tested this again. The script just fails later and not immediately.

Where? Please specify.

**Pull request merged**